### PR TITLE
Use aspnet as runtime

### PR DESCRIPTION
**Pull request by gleono**
This is a fix for #87 

While building the image I noticed that the flag `-r linux-musl-x64` actually bundles all the shared libraries along the app. There is already an image provided by microsoft that contains all these libraries so this is redundant and is platform specific.

The images used as base are actually manifest lists and the correct image will be pulled depending on the architecture of the machine building it. I am relying on the pipeline to test the changes, since I don't have access to a x64 machine.

This change will decrease the image size by about 40% and enable building this on x64 and arm64.

### CODE REVIEW

**Comment by dgrove-oss** [comment]
Thanks for the contribution!  

It looks like the CI passed on linux/x86_64 so it should be fine,  but I'm going to hold off merging until @shawnallen85 has a chance to review.

File: core/net6.0/Dockerfile
Comment (gleono) [inline core/net6.0/Dockerfile:40]: I forgot to mention that this is getting rid of a deprecation warning while building the image.

File: core/net6.0/Dockerfile
Comment (shawnallen85) [inline core/net6.0/Dockerfile:31]: Can you confirm if changing this is necessary? Going from runtime to aspnet does increase the image size by ~ 20%.

Comment (shawnallen85) [reply core/net6.0/Dockerfile:31]: For transparency ...

mcr.microsoft.com/dotnet/aspnet: 117 MB
mcr.microsoft.com/dotnet/runtime: 93.8 MB

Comment (gleono) [reply core/net6.0/Dockerfile:31]: Apologies, I needed to be more clear about the images that I was referring to.

I am comparing the current [openwhisk/action-dotnet-v6.0:1.18.0](https://hub.docker.com/layers/openwhisk/action-dotnet-v6.0/1.18.0/images/sha256-7b6275f129d2aeea3856bb7b769ff3e2f24220b23affa52d473b8f8f33967a6b?context=explore) (76.12MB) and the one [from this change](https://hub.docker.com/layers/gleonoliva/action-dotnet-v6.0/slim/images/sha256-9de9c2a172c4bfa0935b993bd997b063105f1d70fdf379d71f79b13104656abc?context=explore) (42.94 MB). I know that these are different architectures but I expect similar results for amd64.

This is because the step that copies the output from `dotnet publish -r` adds a bunch of libraries that are already in the `aspnet` image, under the `/usr/share/dotnet/shared/` directory to be specific. 

Here is a screenshot of the files bundled in the current image (1.18.0) under the `/app` directory.

image

Once I saw the dlls with "AspNetCore" in the name, I figured that maybe it would be better to switch to aspnet as the runtime image.

Comment (shawnallen85) [reply core/net6.0/Dockerfile:31]: I understood the app size differences, I'm not sold on switching the image yet. Lemme run a comparison on my side, hopefully today, to see if the image needs to be aspnet or if we can stick with runtime.

Comment (gleono) [reply core/net6.0/Dockerfile:31]: Sure thing, I am not sure if this is the way to go for the project since I'm just getting involved with it, it just made sense to me. Hopefully we can find an approach that enables building on M1.
Thanks for taking the time to review.

Comment (shawnallen85) [reply core/net6.0/Dockerfile:31]: So I'm having no luck running the complete test suite locally (podman) and I'm not having any luck on my arm64 (Turing RK1) device. So I cannot test this out locally. Would you be able to change the image on your PR and see if the automated build tests pass?

Comment (gleono) [reply core/net6.0/Dockerfile:31]: Do you mean changing it back to `runtime`? I can do that.

Comment (gleono) [reply core/net6.0/Dockerfile:31]: I pushed the change but I cannot run the pipeline.

Comment (dgrove-oss) [reply core/net6.0/Dockerfile:31]: I approved running it.

**Comment by dgrove-oss** [comment]
Tests failed.   I'll rerun once just to make sure it wasn't a fluke.

**Comment by dgrove-oss** [comment]
All the tests failed second time too (https://github.com/apache/openwhisk-runtime-dotnet/actions/runs/11077953776/job/30788982837?pr=88).  I'd interpret that as meaning we can't use the runtime container without the `-r`

**Comment by shawnallen85** [comment]
Yeah, it looks like using aspnet is the way to go. Those two changes will work with it running on arm64 and amd64. @gleono can you set the image back to aspnet? Sorry for the back and forth on it.

Once that is merged, I can look into seeing what can be done to getting it to build for both arm64 and amd64.

**Comment by gleono** [comment]
No worries, I understand. I was hesitant about the change too since I haven't seen the runtime identifier flag `-r` being used before. I found in the [Microsoft documentation](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish) something that explains why it works, the `dotnet publish --self-contained` section mentions:

> Publishes the .NET runtime with your application so the runtime doesn't need to be installed on the target machine. Default is true if a runtime identifier is specified and the project is an executable project (not a library project).

The Minimal.csproj is indeed an executable, so the behavior was to bundle both the .Net runtime and the ASP.Net framework along the `/app`. Once the flag was removed the framework is now missing, resulting in the majority of the tests failing.

I'm excited about having this built for both architectures, let me know if there is anything I can help with.

**Review by shawnallen85 — APPROVED** [review]

**Pull request merged**
